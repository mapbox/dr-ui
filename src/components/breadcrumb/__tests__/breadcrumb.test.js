import renderer from 'react-test-renderer';
import { testCases } from './breadcrumb-test-cases.js';

describe('Breadcrumb', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.subSite.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.subSite;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.domain.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.domain;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.hide.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.hide;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
