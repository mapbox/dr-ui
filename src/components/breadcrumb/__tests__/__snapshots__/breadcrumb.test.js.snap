// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb Basic renders as expected 1`] = `
<div
  className="dr-ui--breadcrumb "
  data-swiftype-index="false"
>
  <span
    className="none inline-block-mm"
  >
    <a
      className="link"
      href="/mapbox-gl-js/"
    >
      Mapbox GL JS
    </a>
    <span
      className="color-gray-light inline-block-mm none px6"
    >
      <svg
        className="events-none icon inline-block align-t"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      >
        <use
          xlinkHref="#icon-chevron-right"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
  </span>
  <span
    className=""
  >
    <span
      className="color-gray-light inline-block none-mm pr6"
    >
      <svg
        className="events-none icon inline-block align-t"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      >
        <use
          xlinkHref="#icon-arrow-left"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
    <a
      className="link"
      href="/mapbox-gl-js/examples/"
    >
      Examples
    </a>
    <span
      className="color-gray-light inline-block-mm none px6"
    >
      <svg
        className="events-none icon inline-block align-t"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      >
        <use
          xlinkHref="#icon-chevron-right"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
  </span>
  <span
    className="color-gray none inline-block-mm"
  >
    Display a map
  </span>
</div>
`;

exports[`Breadcrumb Example of Platform docs site renders as expected 1`] = `
<div
  className="dr-ui--breadcrumb "
  data-swiftype-index="false"
>
  <span
    className="none inline-block-mm"
  >
    <a
      className="link"
      href="https://platform.mapbox.com"
    >
      Platform
    </a>
    <span
      className="color-gray-light inline-block-mm none px6"
    >
      <svg
        className="events-none icon inline-block align-t"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      >
        <use
          xlinkHref="#icon-chevron-right"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
  </span>
  <span
    className="none inline-block-mm"
  >
    <a
      className="link"
      href="/developer-experience/ci-systems/"
    >
      CI Systems
    </a>
    <span
      className="color-gray-light inline-block-mm none px6"
    >
      <svg
        className="events-none icon inline-block align-t"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      >
        <use
          xlinkHref="#icon-chevron-right"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
  </span>
  <span
    className=""
  >
    <span
      className="color-gray-light inline-block none-mm pr6"
    >
      <svg
        className="events-none icon inline-block align-t"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      >
        <use
          xlinkHref="#icon-arrow-left"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
    <a
      className="link"
      href="/developer-experience/ci-systems/reference/"
    >
      Reference
    </a>
    <span
      className="color-gray-light inline-block-mm none px6"
    >
      <svg
        className="events-none icon inline-block align-t"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      >
        <use
          xlinkHref="#icon-chevron-right"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
  </span>
  <span
    className="color-gray none inline-block-mm"
  >
    Managing credentials
  </span>
</div>
`;

exports[`Breadcrumb Has subsite renders as expected 1`] = `
<div
  className="dr-ui--breadcrumb "
  data-swiftype-index="false"
>
  <span
    className="none inline-block-mm"
  >
    <a
      className="link"
      href="/android/"
    >
      Android
    </a>
    <span
      className="color-gray-light inline-block-mm none px6"
    >
      <svg
        className="events-none icon inline-block align-t"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      >
        <use
          xlinkHref="#icon-chevron-right"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
  </span>
  <span
    className="none inline-block-mm"
  >
    <a
      className="link"
      href="/android/maps/overview/"
    >
      Maps SDK
    </a>
    <span
      className="color-gray-light inline-block-mm none px6"
    >
      <svg
        className="events-none icon inline-block align-t"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      >
        <use
          xlinkHref="#icon-chevron-right"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
  </span>
  <span
    className=""
  >
    <span
      className="color-gray-light inline-block none-mm pr6"
    >
      <svg
        className="events-none icon inline-block align-t"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      >
        <use
          xlinkHref="#icon-arrow-left"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
    <a
      className="link"
      href="/android/maps/overview/"
    >
      Overview
    </a>
    <span
      className="color-gray-light inline-block-mm none px6"
    >
      <svg
        className="events-none icon inline-block align-t"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 18,
            "width": 18,
          }
        }
      >
        <use
          xlinkHref="#icon-chevron-right"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </span>
  </span>
  <span
    className="color-gray none inline-block-mm"
  >
    Camera
  </span>
</div>
`;

exports[`Breadcrumb Hide breadcrumb if there is only one item renders as expected 1`] = `null`;
